modules/export_pipeline/main.tf

```diff
@@ -0,0 +1,20 @@
+data "google_project" "project" {
+}
+
+# Create a GCS bucket for storing source code of cloud functions.
+resource "google_storage_bucket" "source" {
```

Comment:
I think I'd prefer to have a single bucket for the cloud function source.
One thing we could do is define the bucket in `dev/main.tf` and `prod/main.tf` then pass that bucket as a variable into the `data_pipeline` and `export_pipeline` variables. That should work by create a variable of type [object](https://developer.hashicorp.com/terraform/language/values/variables#object) for the bucket.